classes/items/itembase.cpp

```diff
@@ -1,8 +1,62 @@
 /*
-* ITP-298 Capstone Project
-* Group: Dangling Pointers
-* Macarthur Inbody && Nathaniel Mullins
-
-*
+ * ITP 298 Capstone Project (itembase.cpp)
+ * Programmers: Nathaniel Mullins && Macarthur Inbody
+ * Group: Dangling Pointers
+ * Date: 4/5/21
+ * Purpose: Base Item class for our currently Untitled RPG. This is the building block for our weapon and armor classes
 */
+
+#include <sstream>
 #include "itembase.hxx"
+
+//Private functions
+void Item::generate()
+{
+
+}
+
+
+
+//Constructors / Destructors
+Item::Item()
+{
+    std::string name;
+    unsigned type;
```

Comment:
**Optional Changes Can safely ignore**
int is implied by just using unsigned but verbosity helps make intention clearer to view.

Since I for example am using unsigned char for the player's level and all unsigned ints are declared as such. This is a minor thing. It is fine the way it is though as it is valid.

Reply:
### Improvements for future class creations.
Also the constructor should have the option of initializing all of the parameters. 
An easy way to do this and let them also not have to include anything is to do something like.
```
Item::Item(std:string name="", unsigned int type=0, unsigned value=0)
```
then you'd do something like this.
```
this->name = name;
this->type = type;
this->value = value;
```

This way you're still allowing for a Item() construction but also allow them to initialize the object itself with the parameters at the start thus allowing for it's creation without having to go through setters after creation.